lib/UndoInstCombinePass.cpp

```diff
@@ -224,3 +225,30 @@ bool clspv::UndoInstCombinePass::UndoWideVectorShuffleCast(Instruction *inst) {
 
   return true;
 }
+
+bool clspv::UndoInstCombinePass::UndoGEPRemove(Instruction *inst) {
+  auto load = dyn_cast<LoadInst>(inst);
+  if (!load)
+    return false;
+
+  auto gv = dyn_cast<GlobalVariable>(load->getPointerOperand());
```

Comment:
This seems more restrictive than necessary. I wonder whether we should leverage `InferType()` here.

I was playing around with a toy kernel:
```
kernel void zero_copy(global int4* in, global int* out) {
  *out = in[0].x;
}
```

The IR prior to the SPIRVProducer looks like:
```
target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
target triple = "spir-unknown-unknown"

define dso_local spir_kernel void @zero_copy(ptr addrspace(1) nocapture readonly align 16 %in, ptr addrspace(1) nocapture writeonly align 4 %out) local_unnamed_addr #0 !kernel_arg_addr_space !5 !kernel_arg_access_qual !6 !kernel_arg_type !7 !kernel_arg_base_type !8 !kernel_arg_type_qual !9 !clspv.pod_args_impl !10 {
entry:
  %0 = call ptr addrspace(1) @_Z14clspv.resource.0(i32 0, i32 0, i32 0, i32 0, i32 0, i32 0, { [0 x <4 x i32>] } zeroinitializer)
  %1 = getelementptr { [0 x <4 x i32>] }, ptr addrspace(1) %0, i32 0, i32 0, i32 0
  %2 = call ptr addrspace(1) @_Z14clspv.resource.1(i32 0, i32 1, i32 0, i32 1, i32 1, i32 0, { [0 x i32] } zeroinitializer)
  %3 = getelementptr { [0 x i32] }, ptr addrspace(1) %2, i32 0, i32 0, i32 0
  %4 = load <4 x i32>, ptr addrspace(1) %1, align 16
  %5 = extractelement <4 x i32> %4, i64 0
  store i32 %5, ptr addrspace(1) %3, align 4
  ret void
}
```

If I run instcombine on this IR it will drop the geps. ReplacePointerBitcasts will choke on the result because it can't calculate the bitwidth of the runtime array properly to figure out the number of iterations properly. So if we can identify this case generally it would be very helpful.